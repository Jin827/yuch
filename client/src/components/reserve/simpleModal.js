import React from 'react';
import Modal from '@material-ui/core/Modal';
/* --- Components --- */
import Pickers from './pickers';
import TextFields from './textFields';

const SimpleModal = ({ show, reserveInfo, handleClose, handleChange }) => (
  <Modal
    aria-labelledby="simple-modal-title"
    aria-describedby="simple-modal-description"
    open={show}
    onClose={handleClose}
  >
    <div className="tc modal">
      <h3 variant="title" id="modal-title" className="mb2">
        Reservation
      </h3>
      <p>
        예약 상담과 확정을 위해 24시간 이내로 연락을 드리겠습니다. 만약 연락을
        못받으시면, 유청으로 연락주시길 바랍니다.
      </p>
      <div className="reserve-form--box tc">
        <TextFields reserveInfo={reserveInfo} handleChange={handleChange} />
        <Pickers handleChange={handleChange} />
      </div>
      <div>
        <button type="button" className="reserve-btn">
          예약완료
        </button>
        <button type="button" className="reserve-btn" onClick={handleClose}>
          닫기
        </button>
      </div>
    </div>
  </Modal>
);

export default SimpleModal;
