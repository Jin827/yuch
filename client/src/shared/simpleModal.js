import React from 'react';
import Modal from '@material-ui/core/Modal';
import { connect } from 'react-redux';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  paper: {
    position: 'absolute',
    width: theme.spacing.unit * 33,
    backgroundColor: theme.palette.background.paper,
    boxShadow: theme.shadows[5],
    padding: theme.spacing.unit * 6,
    [theme.breakpoints.up('md')]: {
      width: theme.spacing.unit * 36,
      paddingLeft: theme.spacing.unit * 12,
      paddingRight: theme.spacing.unit * 12,
    },
  },
});

const SimpleModal = ({ showModal, classes, component }) => (
  <React.Fragment>
    {showModal && (
      <Modal
        aria-labelledby="simple-modal-title"
        aria-describedby="simple-modal-description"
        open={showModal}
      >
        <div
          className={`tc modal ${classes.paper}`}
          style={{
            top: '50%',
            left: '50%',
            transform: 'translate(-50%, -50%)',
          }}
        >
          {component}
        </div>
      </Modal>
    )}
  </React.Fragment>
);
const mapStateToProps = state => ({ showModal: state.modal.show });

export const Unwrapped = SimpleModal;
export default connect(
  mapStateToProps,
  undefined,
)(withStyles(styles)(SimpleModal));
