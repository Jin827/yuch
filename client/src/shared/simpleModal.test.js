import React from 'react';
import { shallow } from 'enzyme';
import { Unwrapped as UnwrappedSimpleModal } from './simpleModal';

describe('<SimpleModal />', () => {
  const props = {
    showModal: false,
    classes: { paper: {} },
  };
  const wrapper = shallow(<UnwrappedSimpleModal {...props} />);

  it('should not hide the modal when showModal is false', () => {
    expect(wrapper.find('WithStyles(Modal)').exists()).toBeFalsy();
  });

  describe('should not show the modal when showModal is true', () => {
    beforeEach(() => {
      wrapper.setProps({ showModal: true, component: () => {} });
    });

    it('should contain Modal', () => {
      expect(wrapper.find('WithStyles(Modal)').exists()).toBeTruthy();
      expect(wrapper).toMatchSnapshot();
    });

    it('renders correct className', () => {
      expect(
        wrapper
          .find('WithStyles(Modal)')
          .first()
          .props().open,
      ).toEqual(true);
    });
  });
});
