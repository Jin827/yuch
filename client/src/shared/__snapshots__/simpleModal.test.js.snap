// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleModal /> should not show the modal when showModal is true should contain Modal 1`] = `
<Fragment>
  <WithStyles(Modal)
    aria-describedby="simple-modal-description"
    aria-labelledby="simple-modal-title"
    open={true}
  >
    <div
      className="tc modal [object Object]"
      style={
        Object {
          "left": "50%",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
        }
      }
    >
      <Component />
    </div>
  </WithStyles(Modal)>
</Fragment>
`;
